cdci_data_analysis/flask_app/dispatcher_query.py

```diff
@@ -481,89 +473,165 @@ def inspect_state(app):
+                                ctime=os.stat(scratch_dir).st_ctime,
+                                session_id=r.group('session_id'),
+                                job_id=scratch_dir_job_id,
+                                request_completed=request_completed,
+                                aliased_marker=r.group('aliased_marker'),
+                                **result_content
+                            ))
                 else:
                     logger.warning(f"scratch_dir {scratch_dir} not existing, cannot be inspected")
 
-        logger.info("found records: %s", len(records))
+        logger.info("found %s records", len(records_content))
 
-        # TODO adaption to the QueryOutJSON schema is needed
-        return jsonify(dict(records=records))
+        return dict(records=records_content)
 
     @staticmethod
-    def read_scratch_dir(scratch_dir, include_session_log=False):
-        result = {}
+    def read_analysis_parameters_scratch_dir(scratch_dir, decode_token=False):
+        analysis_parameters_obj = None
+        reading_output_message = ""
+        fn = os.path.join(scratch_dir, 'analysis_parameters.json')
+        try:
+            with open(fn) as analysis_parameters_file:
+                analysis_parameters_obj = json.load(analysis_parameters_file)
+        except Exception as e:
+            logger.warning('unable to read: %s', fn)
+            reading_output_message = f'problem reading {fn}: {repr(e)}'
 
+        if analysis_parameters_obj is not None and  decode_token and 'token' in analysis_parameters_obj:
+            analysis_parameters_obj['token'] = tokenHelper.get_decoded_token(analysis_parameters_obj['token'], secret_key=None, validate_token=False)
+
+        return analysis_parameters_obj, reading_output_message
+
+    @staticmethod
+    def read_job_status_scratch_dir(scratch_dir, include_session_log=False, include_status_query_output=False, exclude_analysis_parameters=True):
+        result_job_status = dict(
+            request_completed = False,
+            scratch_dir_fn = scratch_dir,
+            scratch_dir_content = None
+        )
+
+        result_job_status['scratch_dir_content'], result_job_status['request_completed'] = (
+            InstrumentQueryBackEnd.read_content_scratch_dir(scratch_dir,
+                                                            include_session_log=include_session_log,
+                                                            include_status_query_output=include_status_query_output,
+                                                            exclude_analysis_parameters=exclude_analysis_parameters))
+
+        if (not exclude_analysis_parameters and
+                (isinstance(result_job_status['scratch_dir_content']['analysis_parameters'], dict) and
+                 'token' in result_job_status['scratch_dir_content']['analysis_parameters'])):
+            # TODO I am not 100% sure this is enough, perhaps it's not even needed
```

Comment:
I am not sure what do you mean?
We need a check for expired token. This might be ok here.

Reply:
I wasn't sure whether checking against the current time would be enough, and instead using another "time", like the time of the last folder access.

I did try that it wasn't really reliable

That's why I was asking if you had some better suggestions

Reply:
We want to know if there is a chance the job can still finish, as of at the time of the request.
So current time is the right one.